master/buildbot/reporters/github.py

```diff
@@ -146,3 +154,95 @@ def send(self, build):
                     'Fail to send status "{state}" for '
                     '{repoOwner}/{repoName} at {sha}'.format(
                         state=state, repoOwner=repoOwner, repoName=repoName, sha=sha))
+
+
+class GitHubCommentPush(GitHubStatusPush):
+    name = "GitHubCommentPush"
+    neededDetails = dict(wantProperties=True)
+
+    def setDefaults(self, context, startDescription, endDescription):
+        self.startDescription = startDescription
+        self.endDescription = endDescription or 'Build done.'
+
+    def createStatus(self,
+                     repo_user, repo_name, issue, state, description=None):
+        """
+        :param repo_user: GitHub user or organization
+        :param repo_name: Name of the repository
+        :param issue: Pull request number
+        :param state: one of the following 'pending', 'success', 'error'
+                      or 'failure'.
+        :param description: Short description of the status.
+        :return: A deferred with the result from GitHub.
+
+        This code comes from txgithub by @tomprince.
+        txgithub is based on twisted's webclient agent, which is much less reliable and featureful
+        as txrequest (support for proxy, connection pool, keep alive, retry, etc)
+        """
+        payload = {'body': description}
+
+        return self._http.post(
+            '/'.join(['/repos', repo_user, repo_name, 'issues', issue, 'comments']),
+            json=payload)
+
+    @defer.inlineCallbacks
+    def send(self, build):
```

Comment:
I think you can factorize a bit more so that you only need to override createStatus (even if you ignore some of the params)